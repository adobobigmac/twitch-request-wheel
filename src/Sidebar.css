#sidebar {
    margin-left: 12px;
    text-transform: capitalize;
    width: 33%;
}
#sidebar.game {
    width: 33%;
}
#sidebar.seat {
    width: 20%;
}
#sidebar .sidebar-panel {
    background-color: darkslategrey;
    border-radius: 5px;
    margin-top: 0;
    margin-bottom: 10px;
    padding: 1px;
}
#sidebar .sidebar-panel:last-child {
    padding: 5px;
}

#sidebar .sidebar-panel-up-next,
#sidebar .sidebar-panel-history {
    font-size: 14px;
    font-weight: 700;
}
#sidebar .sidebar-panel-up-next {
    padding: 8px;
}
#sidebar .next-game-name {
    margin-bottom: 5px;
}
#sidebar .next-game-party-pack {
    font-size: 11px;
    font-weight: 400;
    margin-top: 0px;
    color: lightgray;
}
#sidebar .sidebar-panel-up-next button {
    font-weight: bold;
    font-size: 12px;
    padding: 5px;
    background-color: aquamarine;
    color: black;
    border-radius: 5px;
    width: 90%;
}
#sidebar .sidebar-panel-up-next button:disabled {
    color: gray;
    font-weight: inherit;
    font-style: italic;
}

#sidebar .nav-wrapper {
    display: flex;
    justify-content: space-between;
}
#sidebar .nav-wrapper button {
    background-color: darkcyan;
    border-radius: 5px;
    margin-top: 0;
    width: 50%;
    margin-bottom: 20px;
}
#sidebar .nav-wrapper button.move-next-game-back {
    margin-right: 5px;
}
#sidebar .nav-wrapper button.move-next-game-fwd {
    margin-left: 5px;
}

#sidebar .sidebar-panel-history-list {
    font-size: 12px;
}
#sidebar .sidebar-panel-history-list li b {
    color: aquamarine;
}
