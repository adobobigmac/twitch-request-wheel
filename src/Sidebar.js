import { Component } from 'react';

export default class Sidebar extends Component {
  constructor(props){
    super(props);
    this.state = {
      index: 0
    }
  }

  printGame = (idx) => {
    if(idx === this.props.nextGameIdx) {
      return (<b style={{color:'aquamarine'}}>{this.props.history[idx]}</b>)
    } else {
      return this.props.history[idx]
    }
  }

  render() {
    return (
      <div style={{marginLeft: "12px", width: "33%", textTransform: 'capitalize'}}>
        <div style={{backgroundColor: "darkslategrey", borderRadius: "5px", marginTop: 0, padding: '1px', marginBottom: '10px'}}>
          <p style={{fontSize: "14px", fontWeight: "700", height: '70px', padding: '8px'}}> Up Next:
            {this.props.history.length > this.props.nextGameIdx
               ? <p>{this.props.history[this.props.nextGameIdx]}</p>
               : <p>not yet decided</p>}
          </p>
        </div>

        <div style={{display: "flex", justifyContent: "space-between"}}>
          <button onClick={this.props.moveNextGameBack} style={{backgroundColor: "darkcyan", borderRadius: "5px", marginTop: 0, width: "50%", marginBottom: '20px', marginRight: '5px'}}> ðŸ¢  </button>
          <button onClick={this.props.moveNextGameFwd} style={{backgroundColor: "darkcyan", borderRadius: "5px", marginTop: 0, width: "50%", marginBottom: '20px', marginLeft: '5px'}}> ðŸ¢¡ </button>
        </div>

        <div style={{backgroundColor: "darkslategrey", borderRadius: "5px", marginTop: 0, padding: '5px'}}>
          <p style={{fontSize: "14px", fontWeight: "700"}}> History </p>
          <p style={{fontSize: "12px"}}>
            {this.props.history.map((playedGame, i) => <li key={i}>{this.printGame(i)}</li> )}
            {this.props.history.length === 0 && <li key='0'>No games yet</li> }
          </p>
        </div>
      </div>
    )
  }
}
