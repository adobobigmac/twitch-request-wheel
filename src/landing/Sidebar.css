#sidebar {
    margin-left: 12px;
    text-transform: capitalize;
    width: 33%;
}
#sidebar.game {
    width: 33%;
}
#sidebar.seat {
    width: 20%;
}
#sidebar .card {
    background-color: darkslategrey;
}
#sidebar .next-game-party-pack {
    font-size: 11px;
    font-weight: 400;
    margin-top: 0px;
    color: lightgray;
}
button#sidebar-up-next {
    font-weight: bold;
    font-size: 12px;
    padding: 5px;
    background-color: aquamarine;
    color: black;
    border-radius: 5px;
    width: 90%;
}
button#sidebar-up-next:disabled {
    color: gray;
    font-weight: inherit;
    font-style: italic;
}

#sidebar .nav-wrapper {
    display: flex;
    justify-content: space-between;
}
#sidebar .nav-wrapper button {
    background-color: darkcyan;
    border-radius: 5px;
    margin-top: 0;
    width: 50%;
    margin-bottom: 20px;
}
#sidebar .nav-wrapper button.move-next-game-back {
    margin-right: 5px;
}
#sidebar .nav-wrapper button.move-next-game-fwd {
    margin-left: 5px;
}

#sidebar .sidebar-history-list-panel li {
    line-height: 1.25;
    font-size: 13px;
    list-style: none;
}
#sidebar .sidebar-history-list-panel li b {
    color: aquamarine;
    font-size: 14px;
}
