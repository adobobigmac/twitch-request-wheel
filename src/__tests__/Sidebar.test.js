import Sidebar from '../Sidebar';
import React from 'react';
import {render, screen} from '@testing-library/react';
import userEvent from '@testing-library/user-event'


describe('Sidebar', () => {

    describe('constructor', () => {
        test('should return the base state object', () => {
            let component = new Sidebar();
            let state = component.state;

            expect(state.index).toBe(0);
        });
    });

    describe('printGame', () => {
        const component = new Sidebar({
            history: [
                {name: 'blather'},
                {name: 'tmp2'}
            ],
            nextGameIdx: 1
        });
        test('should return the game name as-is', () => {
            let game = component.printGame(0);

            expect(game).toBe('blather');
        });
        test('should return the game name rendered with bold tags', () => {
            let game = component.printGame(1);

            expect(game.type).toBe('b');
            expect(game.props.children).toBe('tmp2');
        });
    });

    describe('render', () => {
        test('renders empty page', () => {
            let props = {
                history: [],
                nextGameIdx: 0,
                changeNextGameIdx: ()=>{},
                togglePlayerSelect: ()=>{},
                requestMode: null,    // valid responses: 'seat', 'game'
            }
            render(<Sidebar {...props} />);
            const headerUpNext = screen.getByText(/Up Next:/i);
            expect(headerUpNext).toBeInTheDocument();
        });
        test('renders empty game page', () => {
            let props = {
                history: [],
                nextGameIdx: 0,
                changeNextGameIdx: ()=>{},
                togglePlayerSelect: ()=>{},
                requestMode: 'game',
            }
            render(<Sidebar {...props} />);
            const headerUpNext = screen.getByText(/Up Next:/i);
            expect(headerUpNext).toBeInTheDocument();
        });
        test('renders empty seat page', () => {
            let props = {
                history: [],
                nextGameIdx: 0,
                changeNextGameIdx: ()=>{},
                togglePlayerSelect: ()=>{},
                requestMode: 'seat',
            }
            render(<Sidebar {...props} />);
            const headerUpNext = screen.getByText(/Up Next:/i);
            expect(headerUpNext).toBeInTheDocument();
        });
        test('renders page with content', () => {
            let props = {
                history: [{
                    name: 'Blather \'Round',
                    partyPack: 'Party Pack 7'
                }, {
                    name: 'Trivia Murder Party 2',
                    partyPack: 'Party Pack 6'
                }, {
                    name: 'Survive The Internet',
                    partyPack: 'Party Pack 4'
                }],
                nextGameIdx: 1,
                changeNextGameIdx: jest.fn(),
                moveNextGameFwd: jest.fn(),
                moveNextGameBack: jest.fn(),
                togglePlayerSelect: jest.fn(),
                requestMode: 'game',    // valid responses: 'seat', 'game'
            };
            const {container} = render(<Sidebar {...props} />);
            const headerUpNext = screen.getByText(/Up Next:/i);
            expect(headerUpNext).toBeInTheDocument();
            userEvent.click(container.querySelector('button.move-next-game-fwd'));
            expect(props.changeNextGameIdx).toHaveBeenCalledWith(1);
            userEvent.click(container.querySelector('button.move-next-game-back'));
            expect(props.changeNextGameIdx).toHaveBeenCalledWith(-1);
            expect(props.changeNextGameIdx).toHaveBeenCalledTimes(2);
        });
    });
});
