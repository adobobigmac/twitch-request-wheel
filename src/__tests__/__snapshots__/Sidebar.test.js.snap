// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar render renders page with content 1`] = `
<div>
  <div
    class="game"
    id="sidebar"
  >
    <div
      class="sidebar-panel"
    >
      <div
        class="p sidebar-panel-up-next"
      >
         Up Next:
        <p
          class="next-game-name"
        >
          Trivia Murder Party 2
        </p>
        <p
          class="next-game-party-pack"
        >
          <i>
            Party Pack 6
          </i>
        </p>
        <button
          class="open-seat-requests"
        >
          OPEN SEAT REQUESTS
        </button>
      </div>
    </div>
    <div
      class="nav-wrapper"
    >
      <button
        class="move-next-game-back"
      >
         ⇦ 
      </button>
      <button
        class="move-next-game-fwd"
      >
         ⇨ 
      </button>
    </div>
    <div
      class="sidebar-panel"
    >
      <p
        class="sidebar-panel-history"
      >
         History 
      </p>
      <p
        class="sidebar-panel-history-list"
      >
        <li>
          Blather 'Round
        </li>
        <li>
          <b>
            Trivia Murder Party 2
          </b>
        </li>
        <li>
          Survive The Internet
        </li>
      </p>
    </div>
  </div>
</div>
`;
